qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/dynamic/PartitionJoinTest.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.it.clustering.dynamic;
+
+import static io.camunda.zeebe.it.clustering.dynamic.Utils.assertChangeIsApplied;
+import static io.camunda.zeebe.it.clustering.dynamic.Utils.assertChangeIsCompleted;
+import static io.camunda.zeebe.it.clustering.dynamic.Utils.assertChangeIsPlanned;
+
+import io.camunda.zeebe.management.cluster.PostOperationResponse;
+import io.camunda.zeebe.qa.util.actuator.ClusterActuator;
+import io.camunda.zeebe.qa.util.cluster.TestCluster;
+import java.util.stream.Stream;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+final class PartitionJoinTest {
+  @ParameterizedTest
+  @MethodSource("testScenarios")
+  void canJoinPartition(final Scenario scenario) {
+    // given
+    try (final var cluster = setupCluster(scenario.initialClusterState())) {
+      // when
+      final var response = runOperation(cluster, scenario.operation());
+      // then
+      assertChangeIsPlanned(response);
+      Awaitility.await("Requested change is completed in time")
+          .untilAsserted(() -> assertChangeIsCompleted(cluster, response));
+      assertChangeIsApplied(cluster, response);
```

Comment:
:wrench: Here we should also verify if the final `ClusterTopology` has the broker with newly joined partition.